miio/integrations/ihcooker/tests/test_ihcooker.py

```diff
@@ -0,0 +1,321 @@
+            "000000000000100008005323c111113028006333d121012188007343e130f11000000f93e00141400"
+            "0000f93e001414000000f93e001414000000f93e001414000000f93e001414000000f93e001414000"
+            "000f93e001414000000f93e001414000000f93e001414000000f93e001414000000f93e0014140000"
+            "00f93e001414000000000000000000df87"
+        )
+
+        res = profile_v2.parse(bytes.fromhex(recipe))
+        self.assertEqual(profile_v2.parse(profile_v2.build(res)), res)
+        self.assertEqual(len(profile_v2.build(res)), len(recipe) // 2)
+        self.assertEqual(str(profile_v2.build(res).hex()), recipe)
+
+    def test_phases(self):
+        profile = dict(
+            device_version=4,
+            duration_minutes=115,
+            recipe_id=1002,
+            recipe_name="Test Recipe",
+            menu_location=5,
+            menu_settings=dict(save_recipe=True),
+            stages=[
+                dict(
+                    mode=StageMode.FireMode,
+                    temp_target=60,
+                    temp_threshold=50,
+                    minutes=115,
+                    power=17,
+                    fire_on=19,
+                    fire_off=17,
+                ),
+                dict(
+                    mode=StageMode.TemperatureMode,
+                    temp_target=61,
+                    temp_threshold=51,
+                    minutes=6,
+                    power=18,
+                    fire_on=18,
+                    fire_off=16,
+                ),
+                dict(
+                    mode=StageMode.TempAutoBigPot,
+                    temp_target=62,
+                    temp_threshold=52,
+                    minutes=7,
+                    power=19,
+                    fire_on=17,
+                    fire_off=15,
+                ),
+                dict(
+                    mode=StageMode.TempAutoBigPot,
+                    temp_target=62,
+                    temp_threshold=52,
+                    minutes=7,
+                    power=19,
+                    fire_on=17,
+                    fire_off=15,
+                ),
+            ],
+            crc=0,
+        )
+
+        bytes_recipe = profile_v2.build(profile)
+        hex_recipe = bytes_recipe.hex()
+        self.assertEqual(
+            "030405546573740a52656369706500000000000000000000000000000000000000000003ea8001370000000000000100008137323c111113028006333d121012188007343e130f11188007343e130f11008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414008032f9e52d1414000000000000000000386d",
+            hex_recipe,
+        )
+        self.assertEqual(
+            hex_recipe,
+            profile_v2.build(profile_v2.parse(bytes_recipe)).hex(),
+        )
+        self.assertEqual(115, profile_v2.parse(bytes_recipe).duration_minutes)
+
+    def test_crc(self):
+        recipe = {}
+        self.device.start(recipe)
```

Comment:
That's right, this was helpful during refactoring the codebase. It might also be helpful to clarify what kind of inputs these functions expect. Let me know if you'd like to see anything changed here :-).